include/extractor/scripting_environment_lua.hpp

```diff
@@ -0,0 +1,69 @@
+#ifndef SCRIPTING_ENVIRONMENT_LUA_HPP
+#define SCRIPTING_ENVIRONMENT_LUA_HPP
+
+#include "extractor/scripting_environment.hpp"
+
+#include "extractor/raster_source.hpp"
+
+#include "util/lua_util.hpp"
+
+#include <memory>
+#include <mutex>
+#include <string>
+#include <tbb/enumerable_thread_specific.h>
+
+struct lua_State;
+
+namespace osrm
+{
+namespace extractor
+{
+
+struct LuaScriptingContext final : public ScriptingContext
+{
+    LuaScriptingContext() = default;
+    ~LuaScriptingContext() override = default;
+
+    std::unordered_set<std::string> getNameSuffixList() override;
+    std::vector<std::string> getExceptions() override;
+    void setupSources() override;
```

Comment:
OSRM ties to use the clang naming convention with capitalized member functions.

`void SetupSources()`